lib/Sema/UnresolvedNameTyMapper.hpp

```diff
@@ -57,6 +59,41 @@ class UnresolvedNameTyMapper
                          // gracefully later
         }
 
+        if (auto *structTy = llvm::dyn_cast<glu::types::StructTy>(item)) {
+            auto *params = structTy->getDecl()->getTemplateParams();
+            auto templateArgs = type->getTemplateArgs();
```

Comment:
you need to visit template arguments as well, otherwise `Templated<Str>` will only resolve half: `Templated<UNRESOLVED[Str]>`